src/lib/utils/DY.Lib.SplitFunction.fst

```diff
@@ -0,0 +1,283 @@
+module DY.Lib.SplitFunction
+
+open FStar.List.Tot
+open Comparse //for_allP
```

Comment:
can we move `for_allP` and `for_allP_eq` to somewhere else, so that we do not have to call comparse here?
after all this module is completely unrelated to comparse.
(not a major problem, we can also keep it like this, but then we should add `for_allP_eq` to the comment as well)

Reply:
I agree this is inconvenient. For now I went for the better comment.
I see three possibilities, from best to worst (in my opinion):
- either we try to move `for_allP` in F*'s standard library, which is inconvenient because this requires to do a pull-request
- either we keep using Comparse's `for_allP` because it's already there, which is inconvenient in modules like this one that shouldn't depend on Comparse
- either we re-implement `for_allP` in DY* in some module `DY.Utils` or something, which is inconvenient because this will cause weird shadowing issues with Comparse's `for_allP`

Reply:
I agree, that having `for_allP` (and other basic lemmas) in the F* library would be the best option, but also that it's not the most convenient.
For now, I'd prefer to have a `DY.Utils.List` module in DY* where we can collect basic functions (for example `for_allP`, `memP_map`, ...). I sometimes have to define functions/lemmas on Lists and it would be nice to have a place where we collect all of those.
But we can do this later in a separate step.